### ISSUE TITLE
Rule suggestion: Reserved word used as key

### ISSUE THREAD

**Issue by emilbayes** [eslint/eslint#1144]
The Airbnb Javascript Style Guide disallows [reserved words](https://github.com/airbnb/javascript#objects) to be used as keys in objects. I'd like to enforce this rule. I'm not sure if this is already reported in #667 ("100. Reserved word used as name").

Example (from the style guide):

``` js
// bad
var superman = {
  default: { clark: 'kent' },
  private: true
};

// good
var superman = {
  defaults: { clark: 'kent' },
  hidden: true
};
```

I guess the [complete list](http://es5.github.io/#x7.6.1) would be:  
`break` `do` `instanceof` `typeof` `case` `else` `new` `var` `catch` `finally` `return` `void` `continue` `for` `switch` `while` `debugger` `function` `this` `with` `default` `if` `throw` `delete` `in` `try` `class` `enum` `extends` `super` `const` `export` `import` `implements` `let` `private` `public` `yield` `interface` `package` `protected` `static`
